Sources/Purchasing/Configuration.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Configuration.swift
+//
+//  Created by Joshua Liebowitz on 5/6/22.
+
+import Foundation
+
+/**
+ * ``Configuration`` can be used when configuring the ``Purchases`` instance. It is not required to be used, but
+ * highly recommended. This class follows a builder pattern.
+ *
+ * To configure your `Purchases` instance using this object, follow these steps.
+ *
+ * **Steps:**
+ * 1. Call ``Configuration/configurationBuilder(withAPIKey:)`` To obtain a ``ConfigurationBuilder`` object.
+ * 2. Set this builder's properties using the "`with(`" functions.
+ * 3. Call ``ConfigurationBuilder/build()`` to obtain the `Configuration` object.
+ * 4. Pass the `Configuration` object into ``Purchases/configure(withConfiguration:)``.
```

Comment:
We could even have a swift example here by doing something like 
```
* ```swift
* let configuration = Configuration.configurationBuilder(...
* ...
* ```
*
```

Reply:
✅